diff --git a/src/utils/assets/copy-assets-handler.js b/src/utils/assets/copy-assets-handler.js
index 6b68205d833ce9e8277283ac31230c020d2921ec..2f0a7f018b03eae3b8f3ce1a4cf4790aaafed677 100644
--- a/src/utils/assets/copy-assets-handler.js
+++ b/src/utils/assets/copy-assets-handler.js
@@ -39,12 +39,6 @@ class CopyAssetsHandler {
         this.callback = opts.callback ?? exports.defaultFileEventHandler;
         // TODO(jack): Should handle nested .gitignore files
         this.ignore = (0, ignore_1.default)();
-        const gitignore = pathPosix.join(opts.rootDir, '.gitignore');
-        const nxignore = pathPosix.join(opts.rootDir, '.nxignore');
-        if ((0, node_fs_1.existsSync)(gitignore))
-            this.ignore.add((0, node_fs_1.readFileSync)(gitignore).toString());
-        if ((0, node_fs_1.existsSync)(nxignore))
-            this.ignore.add((0, node_fs_1.readFileSync)(nxignore).toString());
         this.assetGlobs = opts.assets.map((f) => {
             let isGlob = false;
             let pattern;
